pkg/hubble/filters/labels.go

```diff
@@ -29,6 +29,20 @@ func nodeLabels(ev *v1.Event) k8sLabels.Labels {
 	return ciliumLabels.ParseLabelArrayFromArray(labels)
 }
 
+func sourceNodeLabels(ev *v1.Event) k8sLabels.Labels {
+	// For source node labels, we use the same node labels field
+	// as this is the node that observed the flow
+	labels := ev.GetFlow().GetNodeLabels()
+	return ciliumLabels.ParseLabelArrayFromArray(labels)
+}
+
+func destinationNodeLabels(ev *v1.Event) k8sLabels.Labels {
+	// For destination node labels, we use the same node labels field
+	// as this represents the node that observed the flow
+	labels := ev.GetFlow().GetNodeLabels()
+	return ciliumLabels.ParseLabelArrayFromArray(labels)
+}
```

Comment:
I don't think this is correct. You always get the local node labels for both the destination and source node labels. The local node labels will be one of the two, but probably not both.

To be able to do what you want to do, we'd need to add the source and destination labels to the flow, similarly to how Alex did this in https://github.com/cilium/cilium/pull/32851 in `observer/local_node_watcher.go`. However, this time we'd need to get remote node information for the source/destination node.